### ISSUE TITLE
System API and SQLite as default DB

### ISSUE THREAD

**Issue by Abdur-rahmaanJ** [joshiemoore/snakeware#27]
It would be nice to have a system api to expose to app devs.

Lets say someone makes an app to tune volume. Instead of he writing his method, he just uses the system way of changing sound. We don't need everything but at least some system things

Also for storage, was thinking of having sqlite to store system info, using sqlalchemy

**Comment by dargonforce** [comment]
sqlite is a good choice of db for something like this, although it might be slightly overdesigning to use a sql db.  if we do choose to use a sql db then I suggest creating some kind of library (snakesettings or w/e) for accessing and modifying the database, don't leave that up to users.

**Comment by Abdur-rahmaanJ** [comment]
Yes that's the system API. For users to use. We might have a json api, yes to play and get a feel before moving to something serious

**Comment by dargonforce** [comment]
however we end up doing it, it should be implemented in such a way that that users will have no knowledge of the implementation

**Comment by Abdur-rahmaanJ** [comment]
Yaps that's what an API is. Maintainers sweat to let users have a cool experience
